ddtrace/profiling/collector/_task.pyx

```diff
@@ -110,7 +110,7 @@ cpdef list_tasks(thread_id):
             (greenlet_id,
              _threading.get_thread_name(greenlet_id),
              greenlet.gr_frame)
-            for greenlet_id, greenlet in compat.iteritems(_gevent_tracer.greenlets)
+            for greenlet_id, greenlet in list(compat.iteritems(_gevent_tracer.greenlets))
```

Comment:
this probably impacts Python 3 as well ?

any impact to using a list here instead of an iterator, how large can this list get?

Reply:
It really depends. See https://bugs.python.org/issue35615 and all the various bugs around it. 😢 

The list can be as big as the number of greenlet (gevent threads) one can start. In practice, very likely always < 100.

Reply:
Just to be clear: it's not a bug in the profiler code, it's more a bug in the weakref module itself 😢